File: 1894/solution.go
package main

func chalkReplacer(chalk []int, k int) int {
	total := 0
	for _, v := range chalk {
		total += v
	}
	k %= total
	for i, v := range chalk {
		if v > k {
			return i
		}
		k -= v
	}
	return -1
}

File: 1894/solution.ts
function chalkReplacer(chalk: number[], k: number): number {
  let total = 0;
  for (const student of chalk) {
    total += student;
  }
  k %= total;
  for (let i = 0; i < chalk.length; i++) {
    if (chalk[i] > k) {
      return i;
    }
    k -= chalk[i];
  }
  return -1;
}
